src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1505,6 +1512,17 @@ bool AdaptDispatch::HardReset()
     // delete all current tab stops and reapply
     _pConApi->PrivateSetDefaultTabStops();
 
+    // GH#2715 - If all this succeeded, but we're in a conpty, return `false` to
+    // make the state machine propogate this RIS sequence to the connected
+    // terminal application. We've reset our state, but the connected terminal
+    // might need to do more.
+    bool isPty = false;
+    _pConApi->IsConsolePty(isPty);
```

Comment:
This really feels like a function that should have been returning the bool...

Reply:
Big +1. This is one of things I was hoping to clean up when moving to exceptions in place of return values for the error handling (which is still on my long term todo list). But some of these getters aren't even using the return value, so they don't actually need to have the exception handling in place - it should be an easy update.